File: LeetCode-Python/20-有效括号匹配.py
class Solution:
    def isValid(self, s):
        stack = list(['1'])
        for st in s:
            if st in ['(', '[', '{']:
                stack.append(st)
            elif (stack[len(stack)-1]+st) not in ['()', '[]', '{}']:
                stack.append(st)
            else:
                stack.pop()
        if stack == ['1']:
            return True
        else:
            return False

File: LeetCode-Rust/20-有效括号匹配.rs
use std::collections::HashMap;
impl Solution {
    pub fn is_valid(s: String) -> bool {
        let mut stack = vec!["1".to_string()];
        let mut map = HashMap::new();
        map.insert(String::from("{"),"}");
        map.insert(String::from("["),"]");
        map.insert(String::from("("),")");
        for it in s.chars() {
            let c2s = it.to_string();
            let ss = map.get(&c2s);
            if let Some(b) = ss{
                stack.push(c2s.clone());
            } else {
                let mut s1 = stack.pop().unwrap();
                let s2 = map.get(&s1);
                if let Some(U) = s2{
                    if *U != c2s {
                        return false;
                    }
                } else {
                    return false;
                }
            }
        }
        if stack.len()==1{
            return true;
        } else {
            return false;
        }
    }
}
